apps/total-typescript/src/ability/ability.ts

```diff
@@ -52,8 +55,19 @@ const canViewWorkshop = ({user, module, lesson}: ViewerAbilityInput) => {
   //  available to the user (see below)
   const userHasPurchases = Boolean(user && user.purchases.length > 0)
   const hasVideo = Boolean(lesson?.muxPlaybackId)
-
-  return contentIsWorkshop && userHasPurchases && hasVideo
+  const userHasBulkPurchase = Boolean(
+    userHasPurchases && !isEmpty(get(last(user?.purchases), 'bulkCoupon')),
```

Comment:
I don't feel very confident about getting the last item from purchases array tbh.

Reply:
what about checking all of the user's purchases to see if at least one of them has a `bulkCoupon`, e.g. `_.some(user?.purchases || [], (purchase) => !isEmpty(purchase.bulkCoupon))`